src/core/service.c

```diff
@@ -253,10 +255,27 @@ static void service_start_watchdog(Service *s) {
 }
 
 static void service_reset_watchdog(Service *s) {
+        int r;
+        usec_t timeout;
+
         assert(s);
 
         dual_timestamp_get(&s->watchdog_timestamp);
         service_start_watchdog(s);
+
+        if (IN_SET(s->state, SERVICE_START, SERVICE_START_POST)) {
+                timeout = s->timeout_start_usec;
+        } else if (IN_SET(s->state,
```

Comment:
coding style nitpick: not {} around single-line if blocks, also see CODING_STYLE